src/core/mac/mac_frame.hpp

```diff
@@ -1043,328 +1043,365 @@ class Frame : public otRadioFrame
 
     /**
-     * This method returns the key used for frame encryption and authentication (AES CCM).
+     * This method returns a pointer to the Header IE.
      *
-     * @returns The pointer to the key.
+     * @param[in] aIeId  The Element Id of the Header IE.
+     *
+     * @returns A pointer to the Header IE, NULL if not found.
      *
      */
-    const uint8_t *GetAesKey(void) const { return mInfo.mTxInfo.mAesKey; }
+    uint8_t *GetHeaderIe(uint8_t aIeId)
+    {
+        return const_cast<uint8_t *>(const_cast<const Frame *>(this)->GetHeaderIe(aIeId));
+    }
 
     /**
-     * This method sets the key used for frame encryption and authentication (AES CCM).
+     * This method returns a pointer to the Header IE.
      *
-     * @param[in]  aAesKey  The pointer to the key.
+     * @param[in] aIeId  The Element Id of the Header IE.
+     *
+     * @returns A pointer to the Header IE, NULL if not found.
      *
      */
-    void SetAesKey(const uint8_t *aAesKey) { mInfo.mTxInfo.mAesKey = aAesKey; }
+    const uint8_t *GetHeaderIe(uint8_t aIeId) const;
+#endif // OPENTHREAD_CONFIG_MAC_HEADER_IE_SUPPORT
 
     /**
-     * This method returns the IEEE 802.15.4 PSDU length.
+     * This method returns the maximum transmission unit size (MTU).
      *
-     * @returns The IEEE 802.15.4 PSDU length.
+     * @returns The maximum transmission unit (MTU).
      *
      */
-    uint16_t GetPsduLength(void) const { return mLength; }
+    uint16_t GetMtu(void) const;
 
     /**
-     * This method sets the IEEE 802.15.4 PSDU length.
+     * This method returns the FCS size.
      *
-     * @param[in]  aLength  The IEEE 802.15.4 PSDU length.
+     * @returns This method returns the FCS size.
      *
      */
-    void SetPsduLength(uint16_t aLength) { mLength = aLength; }
+    uint16_t GetFcsSize(void) const;
 
     /**
-     * This method returns a pointer to the PSDU.
+     * This method returns information about the frame object as an `InfoString` object.
      *
-     * @returns A pointer to the PSDU.
+     * @returns An `InfoString` containing info about the frame.
      *
      */
-    uint8_t *GetPsdu(void) { return mPsdu; }
+    InfoString ToInfoString(void) const;
+
+private:
+    enum
+    {
+        kInvalidIndex  = 0xff,
+        kSequenceIndex = kFcfSize,
+    };
+
+    uint16_t GetFrameControlField(void) const;
+    uint8_t  FindDstPanIdIndex(void) const;
+    uint8_t  FindDstAddrIndex(void) const;
+    uint8_t  FindSrcPanIdIndex(void) const;
+    uint8_t  FindSrcAddrIndex(void) const;
+    uint8_t  FindSecurityHeaderIndex(void) const;
+    uint8_t  SkipSecurityHeaderIndex(void) const;
+    uint8_t  FindPayloadIndex(void) const;
+#if OPENTHREAD_CONFIG_MAC_HEADER_IE_SUPPORT
+    uint8_t FindHeaderIeIndex(void) const;
+#endif
+
+    static uint8_t GetKeySourceLength(uint8_t aKeyIdMode);
+};
 
+/**
+ * This class implements IEEE 802.15.4 MAC rx frame generation and parsing.
```

Comment:
how about replace with: "This class supports received IEEE 802.15.4 MAC frame processing."